jobs/build/buildvm-maint/Jenkinsfile

```diff
@@ -25,20 +25,42 @@ node('openshift-build-1') {
         buildlib.initialize()
 
         sshagent(["openshift-bot"]) {
-            stage("oit setup"){
-                sh 'rm -rf enterprise-images'
-                sh 'git clone git@github.com:openshift/enterprise-images.git'
+
+            // Capture exceptions and don't let one problem stop other cleanup from executing
+            e1 = null
+            e2 = null
+
+            try {
+                stage("oit setup"){
+                    sh 'rm -rf enterprise-images'
+                    sh 'git clone git@github.com:openshift/enterprise-images.git'
+                }
+
+                stage("push images") {
+                    dir ( "enterprise-images" ) {
+                        sh './oit/oit.py --user=ocp-build --group sync-3.7 distgits:push-images --to-defaults'
+                    }
+                }
+            } catch ( ex1 ) {
+                e1 = ex1
```

Comment:
is there not a `finally`?

Reply:
There is. But maintaining something like .. try { X } finally { try { Y } finally { ..Z } } .. as more steps are added isn't a great pattern. Certainly cleaner ways than what I've done here though.